### ISSUE TITLE
 clothing-co-parsing do not generate seg label for test/val dataset?

### ISSUE THREAD

**Issue by chaipangpang** [sangwoomo/instagan#32]
Hi，I find MHP and COCO generate seg label for test/val dataset, but CPP have not.
generate_ccp_dataset_val()  in  generate_ccp_dataset.py do not save seg label.
Why is that？
Who to test the model "python test.py --dataroot ./datasets/jeans2skirt_ccp --model insta_gan --name jeans2skirt_ccp_instagan --loadSizeH 300 --loadSizeW 200 --fineSizeH 300 --fineSizeW 200"?

**Comment by sangwoomo** [comment]
Hi, basically we run both train and test on the train dataset (as the goal of GANs are mimicking the training distribution). Hence, you can simply add option `--phase train` to test the trained model on the train dataset.
However, we also provide validation dataset to check the generalizability of the model. To this end, we train an additional segmentation model on the CCP train dataset, and predicted masks on the CCP val dataset. As CCP dataset contains only one mask per image, we simply used the pix2pix model, which can be simply run under the code base.
Sorry for not providing the pre-trained weights of the segmentation model, but it is not so difficult to reproduce it.

**Comment by chaipangpang** [comment]
I got it, thanks.

**Comment by chaipangpang** [comment]
![test](https://user-images.githubusercontent.com/15702596/63759875-d8695a00-c8f0-11e9-9468-d1250dbdaacb.png)
![train](https://user-images.githubusercontent.com/15702596/63759885-dacbb400-c8f0-11e9-98b9-baeef64f2153.png)

It worked, thanks！
But, I have encountered some new problems while runing test ccp. 
There are two results output during training and testing(mark in the top-left corner of the image), but results  during testing(upper one) seems to be unnormal.
Shouldn’t this result be like the results during training(lower one)? Why are they different?
I checked the code for this part, 'save_images()' and 'display_current_results()' both use 'util.tensor2im()' and save images to html.
What could be a reason for getting this problem? thanks!

**Comment by sangwoomo** [comment]
Hi, what is the upper one and lower one?
The lower one seems to work ok, but do you have a problem at the upper one?

**Comment by chaipangpang** [comment]
I am sorry for the expression is not accurate.
I have re-edited the previous message.
thanks!

**Comment by sangwoomo** [comment]
Thank you for your revised question!
Did you you the model of the same epoch?
Occasionally, the training becomes weirdly unstable, and all the results collapse.
Can you check 1) your loss curves are stable, 2) test results are still weird for earlier epochs?
